### ISSUE TITLE
Increase the radius at launch, when the storage limit increases

### ISSUE THREAD

**Issue by carver** [ethereum/trin#624]
Before #614, the radius was always resetting to 100% at the beginning of every run. This felt wrong when we had already identified the radius on previous runs with the same storage limit, so #614 identifies the radius based on the existing storage.

This works fine when the storage limit stays the same between runs, but when the storage increases on a new run, it means the radius stays too small. The node will sub-optimally avoid storing things that it ought to be interested in storing.

Before and after #614, trin correctly handles when you decrease the storage limit: it will prune down the database and reduce the radius as needed.

Some options:
1. Keep the previous storage limit in the database, to detect if there was a change on launch and adjust the radius
2. Adjust the radius as a percentage increase based on the amount used vs the limit
3. ?

I think # 2 is the best option here

**Comment by pipermerriam** [comment]
My academic opinion on this:

- CLI should accept only one of: `--kb` or `--data-radius`
  - `--kb` means radius is a function of total storage
  - `--data-radius` means that radius is fixed and total storage is infinity.
  
In the event that the client is using `--data-radius`, then storage is effectively unlimited and we operate using a hard-coded radius.  This is un-interesting and should be reasonably easy to implement.

In the event that the client is using `--kb` then we need to handle the following situations.

- `disk_usage() < max_disk_usage()` -> `radius = 100%`
- `disk_usage() >= max_disk_sage()` -> `radius = computed_radius()` and `maybe_evict(storage)`

In the case where we call `maybe_evict()`, I think the eviction logic should be roughly as follows.

```
loop {
    furthest_content_id = get_furthest_content_id()
    furthest_content_size = get_content_size(furthest_content_id);
    if disk_usage() - furthest_content_size >= max_disk_usage() {
        // only evict if we will still be maxed out on storage.
        evict(content_id);
    }  else {
        break;
    }
}
```

The idea is to only evict when our usage **after** the eviction will still be at 100% utilization.  I think this avoids the situation where we have to *treat* storage as being full when actually it isn't quite full, it just *was* full a moment ago before we evicted anything.  I think this approach lets us simply always compute our radius based on the amount of storage the client was configured with and to maximize that storage.

I agree that there's some small amount of in-efficiency in setting radius to 100% when we only have a small amount of capacity, however, I think this is the correct way to go about it and I think it keeps things simple.

**Comment by carver** [comment]
> * `--data-radius` means that radius is fixed and total storage is infinity.

AFAIK, we don't support a `--data-radius` flag, and I'm happy leaving it that way. I know I confused things by saying "user-supplied" sometimes when talking. The starting radius is always a hard-coded default, which is 100% for an empty DB. 



> In the event that the client is using `--kb` then we need to handle the following situations.
> 
>     * `disk_usage() < max_disk_usage()` -> `radius = 100%`
> 
>     * `disk_usage() >= max_disk_sage()` -> `radius = computed_radius()` and `maybe_evict(storage)`

I started in roughly this direction. Here's why I changed: there is roughly no situation in which we start up with *exactly* the storage used. So then in the typical case, you're always resetting to 100% on restart, accepting content way out of range and then evicting.

I see the proposed approach to that is to always overfill the storage a bit. Calculating the relative radius (as in #625) isn't too hard. I'm not sure one is obviously better than the other either way, except that #625 is already built. I'm not convinced that it's worth this change just to add a few kb at the margins on clients.

Is there any other benefit I'm missing?
